src/Graphics/OpenGLContext/GLSL/glsl_CombinerProgramBuilderCommon.h

```diff
@@ -0,0 +1,133 @@
+		const ShaderPart * getFragmentShaderEnd() const override;
+
+		bool isObsolete() const override;
+
+	private:
+
+		GLuint _getVertexShaderRect() const override;
+		GLuint _getVertexShaderTriangle() const override;
+		GLuint _getVertexShaderTexturedRect() const override;
+		GLuint _getVertexShaderTexturedTriangle() const override;
+
+		void _writeSignExtendAlphaC(std::stringstream& ssShader) const override;
+		void _writeSignExtendAlphaABD(std::stringstream& ssShader) const override;
+		void _writeAlphaTest(std::stringstream& ssShader) const override;
+		void _writeSignExtendColorC(std::stringstream& ssShader) const override;
+		void _writeSignExtendColorABD(std::stringstream& ssShader) const override;
+		void _writeClamp(std::stringstream& ssShader) const override;
+		void _writeCallDither(std::stringstream& ssShader) const override;
+		void _writeBlender1(std::stringstream& ssShader) const override;
+		void _writeBlender2(std::stringstream& ssShader) const override;
+		void _writeBlenderAlpha(std::stringstream& ssShader) const override;
+		void _writeLegacyBlender(std::stringstream& ssShader) const override;
+		void _writeFragmentHeader(std::stringstream& ssShader) const override;
+		void _writeFragmentHeaderDither(std::stringstream& ssShader) const override;
+		void _writeFragmentHeaderNoise(std::stringstream& ssShader) const override;
+		void _writeFragmentHeaderWriteDepth(std::stringstream& ssShader) const override;
+		void _writeFragmentHeaderDepthCompare(std::stringstream& ssShader) const override;
+		void _writeFragmentHeaderMipMap(std::stringstream& ssShader) const override;
+		void _writeFragmentGlobalVariablesNotex(std::stringstream& ssShader) const override;
+		void _writeFragmentHeaderCalcLight(std::stringstream& ssShader) const override;
+		void _writeFragmentMain2Cycle(std::stringstream& ssShader) const override;
+		void _writeFragmentMain(std::stringstream& ssShader) const override;
+		void _writeFragmentBlendMux(std::stringstream& ssShader) const override;
+		void _writeShaderCoverage(std::stringstream& ssShader) const override;
+		void _writeFragmentCorrectTexCoords(std::stringstream& ssShader) const override;
+		void _writeFragmentReadTexMipmap(std::stringstream& ssShader) const override;
+		void _writeFragmentCallN64Depth(std::stringstream& ssShader) const override;
+		void _writeFragmentRenderTarget(std::stringstream& ssShader) const override;
+		void _writeShaderFragmentMainEnd(std::stringstream& ssShader) const override;
+		void _writeShaderCalcLight(std::stringstream& ssShader) const override;
+		void _writeShaderNoise(std::stringstream& ssShader) const override;
+		void _writeShaderDither(std::stringstream& ssShader) const override;
+		void _writeShaderWriteDepth(std::stringstream& ssShader) const override;
+		void _writeShaderN64DepthCompare(std::stringstream& ssShader) const override;
+		void _writeShaderN64DepthRender(std::stringstream& ssShader) const override;
+
+		typedef std::unique_ptr<ShaderPart> ShaderPartPtr;
```

Comment:
better
`using ShaderPartPtr = std::unique_ptr<ShaderPart>;`
and place it in public section of glsl::CombinerProgramBuilder class.
the same for CombinerProgramBuilderFast and CombinerProgramBuilderAccurate